docs/infrastructure/vs-deps-branches.md

```diff
@@ -0,0 +1,132 @@
+make the breaking change to the API and we can update our code in a few days. They may have to leave an Obsolete version of the API for a few days while we get our stuff in, but that's on the order of days. Us only moving to
+previews would mean they'd be having to ship already deprecated APIs in previews; and if it stretches across minor versions like before, then across minor versions.
+
+## OK, so we have to check in things sooner. What if we just told our contributors to run preview builds and deal with occasional break?
+
+Right now, we do make things worse on ourselves by generally mandating non-preview builds to contributors. It means the window where a break would go into Roslyn to being resolvable by moving to a new build can be a few months.
+If we required contributors to move to preview builds it would mean the "time until a break is fixed" goes down. However:
+
+1. Sometimes one preview build may be quite a while after another. Because of the holidays, 16.5 Preview 1 and 16.5 Preview 2 shipped pretty far apart -- Preview 1 shipped on December 3rd and Preview 2 shipped
+   on January 22nd, a difference of 50 days.
+2. Even if one preview has gone out, that doesn't mean that master will now work with it. There's usually a two or three week internal stabilization on a preview build, but by then we may have already made further
+   breaking changes in master.
+
+These two factors mean that a break wouldn't be a rare thing.
+
+## OK, so can we avoid breaking changes? Can we use more lightup?
+
+Possibly, and when we can easily do that we already do, but the moment the API gets tricky, we run into a few issues.
+
+First, many components in Visual Studio bump assembly versions every minor release. If we move to consuming packages for 16.5, there aren't binding redirects if you're only running this on
+16.4. It's busted, _even if we didn't consume any new APIs._ Assuming we updated Visual Studio to have binding redirects that redirect _down_ to older versions, we still run into CLR limitations.
+If the API requires an existing type to implement a new interface, the type can't be loaded at all if the interface is missing.
+
+Also, breaking changes aren't always unavoidable. We do sometimes have to do coordinated insertions. Being able to use the -vs-deps branch as a buffer has been helpful.
+
+## What about splitting our build in some other way?
+
+When we were trying to support Dev14 and Dev15 at the same time, Roslyn did a model where we had two VSIXes; there was one that had most of our stuff that ran on Dev14 and Dev15, and one that had the
+new features for Dev15. This did mean that Dev14 was still supportable, but had a bunch of limitations:
+
+1. It still didn't solve dealing with API breaks or changes within the Dev15 subset. That was still an all-or-nothing.
```

Comment:
This sentence somehow doesn't quite carry the weight of the problem it's describing. Once realized they were going to change APIs within the "shipped" subset the splitting between VSIX ideas was pretty much "game over man". 

Basically this point renders the rest a bit moot cause it's such a death nail to this approach.